src/hotspot/share/memory/heap.hpp

```diff
@@ -38,8 +38,8 @@ class HeapBlock {
 
  public:
   struct Header {
-    size_t  _length;                             // the length in segments
-    bool    _used;                               // Used bit
+    uint32_t  _length;                           // the length in segments
+    bool      _used;                             // Used bit
```

Comment:
The `Header` size is aligned up 8 bytes. With `size_t length;` the aligned size will be 16 bytes.

`HeapBlock/Header` is used by CodeCache only which has 2Gb limit. The `length` counts number of segments which smallest size (`CodeCacheSegmentSize`) is 64 bytes.  32 bits is enough to cover it.